More front end cleanup

mikefranze opened the pull request:
Built on top of #176 

Cleans up the component file structures a bit
Adds buttons to flip through different races instead of having them all on one page.
Changed indexing in star ballot to be consistent with scores. Before 0 mean nothing selected and 1-6 mapped to 0-5. Now null means nothing selected and the rest maps as expected.

https://user-images.githubusercontent.com/41272412/196080820-d32efccf-80d7-4fa5-a96f-defcf40e2640.mp4

MagnusMuscle commented on backend/src/Models/CastVoteStore.ts:
```diff
@@ -29,11 +29,12 @@ export default class CastVoteStore {
             ballot.ip_address,
             JSON.stringify(ballot.votes),
-            JSON.stringify(ballot.history),
+            JSON.stringify(ballot.history),,
```
Is this double comma intentional? Seems wrong to me.

mikefranze replied:
Good catch, was a typo

mikefranze commented:
Added a few more big updates that basically touched every file, I don't expect people to really dig through the code for review so I'll include screenshots of the final product. Still needs some styling but this should help make it easier in the future.

Added hooks for local storage so the election forms now persist if you close the window.

Added a random "tempID" to local storage. If the user isn't logged in it will use that as the user ID. The intention is to allow you to submit a quick poll if you're not logged in and also be able to (in a future update) see all polls created by that ID.

Added a quick poll form to the landing page with a button to log in for more features. This uses the "tempID if you're not logged in and sets the election state to "open".
![image](https://user-images.githubusercontent.com/41272412/197435921-26d141ba-a549-4cf3-8398-ce8f344a3087.png)

Upgraded the MUI styling library to the latest version which changed a lot of import lines.

Updated the header to be collapsible and responsive
![image](https://user-images.githubusercontent.com/41272412/197436191-19ebc63a-11fe-4f8e-bb01-aa1cb49d3ff3.png)

Some styling updates to the election form and election landing page
![image](https://user-images.githubusercontent.com/41272412/197436363-0e87c68a-df7c-47f3-9a35-61f484727656.png)
![image](https://user-images.githubusercontent.com/41272412/197436446-5a0fa096-0908-4d94-8c10-030e32f97ec2.png)

And most importantly: Share buttons!
![image](https://user-images.githubusercontent.com/41272412/197436504-1b765bd0-5b3c-4f5d-aa45-200bb933b91b.png)

scottPlusPlus reviewed (commented):
![firefox_aWUWz3M00r](https://user-images.githubusercontent.com/40651157/197660388-7369c66b-ec5b-438c-b4e6-5b5e581d1376.png)

Seems like the logged out version isn't quite working for me?

Also the log-in doesn't seem to be working for me either?  I can get through the keycloak, but then once it redirects me, I see all the sign in data in the url (ex: http://localhost:3000/?session_state=0ca8cbbc-bae2-4e8a-87a3-705cbf4839ed&code=4a4e1e84-dbbe-451c-a515-395d6914bb14.0ca8cbbc-bae2-4e8a-87a3-705cbf4839ed.7804660f-ad64-4892-b022-4229716d1107), but the top-right nav still says "Login"

I assume this is my user-error, but LMK

mikefranze commented:
Found the issue for the quick poll missing the vote button. I had moved it recently under a condition for when the election is open but forgot about another condition. 

For logged in issue try refreshing the page. Currently the log in checks are inside a class object being passed around and when that changes it doesn't trigger a rerender. I'm going to tackle that another day.

scottPlusPlus commented:
The refresh still didn't work for me.  But anyway this is probably a net win.  Let's go ahead and land the refactor and then can work on fixing any issues one at a time.

mikefranze commented:
You might be missing the keycloak public key
